File: LeetCode/01/lc122.c
/* lc122.c */

/* LeetCode 122 Best Time to Buy and Sell Stock II `E` */
/* A~0c04 */
/* acc | 100% | 17' */

int maxProfit(int *prices, int pricesSize)
{
    if (!prices || !pricesSize)
        return 0;

    int profit = 0;
    for (int i = 1; i < pricesSize; i++)
        if (prices[i] > prices[i - 1])
            profit += prices[i] - prices[i - 1];

    return profit;
}

File: LeetCode/01/lc122.py
# lc122.py

# LeetCode 122 Best Time to Buy and Sell Stock II `E`
# A~0c04
# acc | 78%


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) <= 1:
            return 0

        profit = 0
        for i in range(len(prices) - 1):
            if prices[i + 1] > prices[i]:
                profit += prices[i + 1] - prices[i]

        return profit
